### ISSUE TITLE
Show a warning that uses-material-design: true needs to be specified in pubspec.yaml when setting useMaterial3: true for users Migrating a Cupertino app to Material 

### ISSUE THREAD

**Issue by khanhnwin** [flutter/flutter#130935]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

Migrating a Cupertino app to Material and decided that I wanted to use [Material's icons](https://api.flutter.dev/flutter/material/Icons-class.html). Started adding icons into the bottom navigation bar, but only questions marks in a square (BottomNavigationBar) or weird emojis (NavigationBar) appeared. There's no indicator that I need to specify `uses-material-design: true` in my pubspec.yaml.

BottomNavigation:
<img width="447" alt="Screenshot 2023-07-19 at 12 56 53 PM" src="https://github.com/flutter/flutter/assets/12416400/f2c8e898-2385-40c2-94bd-00b1efbdfdb6">

NavigationBar:
<img width="450" alt="Screenshot 2023-07-19 at 12 56 34 PM" src="https://github.com/flutter/flutter/assets/12416400/0b1ed4eb-486e-4b25-a311-5f13ca4afe75">

### Proposal

Show a warning that `uses-material-design: true` needs to be specified in pubspec.yaml when using Material icons.

**Comment by dam-ease** [comment]
Hi @khanhnwin. Thanks for the proposal. Do you mean instead of the question marks in a square, there should be a warning in app?? Because the documentation states this:

> To use this class, make sure you set uses-material-design: true in your project's pubspec.yaml file in the flutter section. This ensures that the Material Icons font is included in your application

as well as inside `pubspec.yaml` file:

>  The following line ensures that the Material Icons font is included with your application, so that you can use the icons in the material Icons class.

**Comment by khanhnwin** [comment]
Hey @dam-ease! Yes, I meant either on the command line or within the app. In this app's case - the pubspec didn't have any of the boilerplate comments that is usually present on `flutter create` - it was originally a cupertino app, so no Material. 

The documentation is helpful, but use-material-design defaults to true on flutter create so if devs are used to building projects from scratch, that's not a setting that they have to think about very often.

**Comment by HansMuller** [comment]
Good suggestion, we should do this.  Ideally the analyzer could detect the incompatibility between pubspec.yaml and the app's dependencies? @goderbauer is that possible? Otherwise we could produce an "actionable" error message at run-time per https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#only-log-actionable-messages-to-the-console

**Comment by goderbauer** [comment]
> Ideally the analyzer could detect the incompatibility between pubspec.yaml and the app's dependencies?

That's a question for the analyzer folks, maybe @pq or @srawlins could answer whether it would be possible to have a lint that reminds you to add `uses-material-design: true` to your pubspec when you use [Material's icons](https://api.flutter.dev/flutter/material/Icons-class.html) in your code.

**Comment by srawlins** [comment]
> whether it would be possible to have a lint that reminds you to add uses-material-design: true to your pubspec when you use [Material's icons](https://api.flutter.dev/flutter/material/Icons-class.html) in your code.

What does "use Material's icons" mean? Is it just a reference to the `Icon` class or constructing it or something? If so, yes that could be a lint rule.

**Comment by HansMuller** [comment]
Right, it just means that the app has a reference to the `Icons` (plural) class - https://api.flutter.dev/flutter/material/Icons-class.html

**Comment by pq** [comment]
Sounds like a useful lint to me.

@khanhnwin: could you open a lint proposal [here](https://github.com/dart-lang/linter/issues/new?assignees=&labels=lint+proposal%2C+status%3A+pending&projects=&template=lint-proposal.md&title=proposal%3A+%60%3Crule_name%3E%60) -- I don't think I can transfer this issue across orgs...
